### PR TITLE
Added API to set offline in NavigationView

### LINKED ISSUE #1285: As an end developer I want to choose between online and offline routing in the Navigation View
The PR that closes this ticket should add a configuration option to the Navigation View and a new public method so the end developer can switch between online and offline routing.

The configuration option should be called something like `offlineRoutingData` and is a file path where the unpacked routing tiles are located.

The public api should be called something like `shouldRouteOffline` and will take a boolean. When `true` is passed the offline router will be used and when `false` is passed the online router will be used. Moving between offline and online should not interupt the current session but when the next routing request is made the correct routing option should be used.

### PR DESCRIPTION
**Pull request by devotaaabel**
Closes #1285 

This PR adds `NavigationView` APIs that will allow initialization of offline data + a `boolean` toggle for switching between online / offline `DirectionsRoute` requests.

### CODE REVIEW

**Comment by mcwhittemore** [comment]
Will this close #1285 @devotaaabel?

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java [hunk 1]
```diff
@@ -477,6 +478,10 @@ private FixLocation buildFixLocationFrom(Point point) {
     );
   }
 
+  public void setOffline(String tileFilePath, String translationsDirPath) {
```
Comment (danesfeder) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java]: Is this API being used?

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java [hunk 2]
```diff
@@ -35,6 +35,7 @@
 import com.mapbox.services.android.navigation.v5.route.FasterRouteListener;
 import com.mapbox.services.android.navigation.v5.routeprogress.ProgressChangeListener;
 import com.mapbox.services.android.navigation.v5.snap.Snap;
+import com.mapbox.services.android.navigation.v5.utils.ValidationUtils;
```
Comment (danesfeder) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java]: Unused import added

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java [hunk 3]
```diff
@@ -443,11 +452,19 @@ private void sendEventArrival(RouteProgress routeProgress, Milestone milestone)
...
+  private void handleOffRouteEvent(Location location) {
+    Point newOrigin = Point.fromLngLat(location.getLongitude(), location.getLatitude());
     if (navigationViewEventDispatcher != null && navigationViewEventDispatcher.allowRerouteFrom(newOrigin)) {
       navigationViewEventDispatcher.onOffRoute(newOrigin);
       OffRouteEvent event = new OffRouteEvent(newOrigin, routeProgress);
-      navigationViewRouteEngine.fetchRouteFromOffRouteEvent(event);
+
+      if (isOffline) {
+        ArrayList<Point> locations = viewRouteFetcher.calculateRemainingCoordinates(event);
+        navigation.findOfflineRouteFor(location, locations);
```
Comment (danesfeder) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java]: It looks like we are finding the route here, but not updating the `DirectionsRoute` with `updateRoute(DirectionsRoute route)` (we do this when we get a route back with `ViewRouteFetcher`.  So the route found will never propagate here

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java [hunk 4]
```diff
@@ -65,7 +66,7 @@
   final MutableLiveData<Boolean> shouldRecordScreenshot = new MutableLiveData<>();
 
   private MapboxNavigation navigation;
-  private ViewRouteFetcher navigationViewRouteEngine;
+  private ViewRouteFetcher viewRouteFetcher;
```
Comment (danesfeder) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java]: thanks for updating this naming 👍

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java [hunk 5]
```diff
@@ -415,6 +415,14 @@ public MapboxNavigation retrieveMapboxNavigation() {
     return navigationViewModel.retrieveNavigation();
   }
 
+  public void initializeOfflineData(String tileFilePath, String translationsDirPath) {
```
Comment (danesfeder) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java]: Let's add javadoc for these new methods here - I added some javadoc in `nav-native-offline` that you can take a look at after rebasing.

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java (hunk 3 above)
Comment (devotaaabel) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java]: @danesfeder thanks! I was worried I was missing something like that

**Comment by devotaaabel** [comment]
I'm closing this until `nav-native-offline` is merged into `navigation-native`

**Pull request closed without merging**